### PR TITLE
Cleanup coverage

### PR DESCRIPTION
**Pull request by ForbesLindesay**
This splits the coverage task out by adding “precoverage” and “postcoverage”.  It also deletes the intermediate coverage files once it is done.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,7 +66,9 @@
   },
   "scripts": {
     "test": "mocha -R spec && npm run coverage",
-    "coverage": "rimraf cov-pt* && istanbul cover --report none --dir cov-pt0 node_modules/mocha/bin/_mocha && istanbul report --include cov-pt\\*/coverage.json",
+    "precoverage": "rimraf coverage && rimraf cov-pt*",
+    "coverage": "istanbul cover --report none --dir cov-pt0 node_modules/mocha/bin/_mocha -- -R dot",
```
Comment (TimothyGu) [inline package.json]: Why `-R dot`?

**Comment by TimothyGu** [comment]
Are you sure the `pre` and `post` work as expected?

File: package.json (hunk 1 above)
Comment (TimothyGu) [inline package.json]: Actually I get why you put it there now. We already have a dedicated `test` script with proper output.

**Comment by ForbesLindesay** [comment]
Yes, "pre" and "post" work for any command (unless that command already starts with "pre" or "post", it's not recursive).  They make for a really nice clean way to do setup and cleanup :)

**Pull request merged**